amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -612,6 +623,51 @@ void cleanupShardConsumers(Set<ShardInfo> assignedShards) {
         }
     }
 
+    /**
+     * Exceptions in the RxJava layer can fail silently unless an error handler is set to propagate these exceptions
+     * back to the KCL, as is done below. This method is internal to the class and has package access solely
+     * for testing.
+     *
+     * @param handler This method accepts a handler param solely for testing. All non-test calls to this method will
+     *                have a null input.
+     */
+    @VisibleForTesting
```

Comment:
Can this be tested as part of runProcessLoop() method? 
We want to make sure this is invoked from runProcessLoop.

Reply:
Changed the unit test for this method to call `RxJavaPlugins.onError()` in runProcessLoop() on a scheduler spy to verify that the error handler works as expected

Reply:
If we get the scheduler take an EventFactory, then we can mock the event vended to the scheduler, and assert that .accept is invoked on the event upon failure. 

That way, this method can be marked private, and no need to support custom handler as an argument to this method.

Reply:
To be able to mock the EventFactory for testability, we would need a public interface for setting/modifying the EventFactory (either in configs or a setter). Currently we don't want users to be able to modify this factory so this would only solve the problem if we introduced a 'backdoor' non-public setter for testing, which still isn't really designing for testability. Unless there's another way to inject a mock that doesn't rely on a public gateway?

Reply:
Refactored to use a factory by overloading with a protected constructor for testing